src/main/java/io/github/mivek/model/WeatherCondition.java

```diff
@@ -89,7 +89,7 @@ public void addPhenomenon(final Phenomenon pPhenomenon) {
      * @return true if there is at least phenomenon.
      */
     public boolean isValid() {
-        return !phenomenons.isEmpty() || (intensity != null && descriptive != null);
+        return !phenomenons.isEmpty() || intensity != null || descriptive != null;

```

Comment:
I do not think this solves the problem. This will lead to token `-` or `+` to be recognized as valid weather conditions.
Plus I am not sure all descriptive can be weather conditions on their own.

I know that `TS` can be coded itself but I do not know about the others.
What we could do is having a set of Descriptive to be considered sufficient to be weather conditions on their own and in the method  `isValid` checking if the descriptive is in the set.

What do you think?

Reply:
mmm true, you are right!

maybe 
```JAVA
        return !phenomenons.isEmpty() || intensity != null || descriptive != null;
```
can be
```JAVA
        return !phenomenons.isEmpty() || !descriptives.isEmpty(); //with more than one descriptive support
```

so the we don't care of intensity for for the validation, is valid if some phenomena or descriptive are found.

It's Ok?

Reply:
Well with "movement" of multiple descriptive field the "isValid" clause must be:
```JAVA
        return !phenomenons.isEmpty() || descriptive != null;
```
in this branch of course :smile: